Extension/src/LanguageServer/Providers/CopilotHoverProvider.ts

```diff
@@ -0,0 +1,129 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All Rights Reserved.
+ * See 'LICENSE' in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+import * as vscode from 'vscode';
+import { Position, ResponseError } from 'vscode-languageclient';
+import * as nls from 'vscode-nls';
+import { DefaultClient, GetCopilotHoverInfoParams, GetCopilotHoverInfoRequest } from '../client';
+import { RequestCancelled, ServerCancelled } from '../protocolFilter';
+import { CppSettings } from '../settings';
+
+nls.config({ messageFormat: nls.MessageFormat.bundle, bundleFormat: nls.BundleFormat.standalone })();
+const localize: nls.LocalizeFunc = nls.loadMessageBundle();
+
+export class CopilotHoverProvider implements vscode.HoverProvider {
+    private client: DefaultClient;
+    private currentDocument: vscode.TextDocument | undefined;
+    private currentPosition: vscode.Position | undefined;
+    private waiting: boolean = false;
+    private ready: boolean = false;
+    private cancelled: boolean = false;
+    private cancelledDocument: vscode.TextDocument | undefined;
+    private cancelledPosition: vscode.Position | undefined;
+    private content: string | undefined;
+    constructor(client: DefaultClient) {
+        this.client = client;
+    }
+
+    public async provideHover(document: vscode.TextDocument, position: vscode.Position, token: vscode.CancellationToken): Promise<vscode.Hover | undefined> {
+        await this.client.ready;
+        const settings: CppSettings = new CppSettings(vscode.workspace.getWorkspaceFolder(document.uri)?.uri);
+        if (settings.hover === "disabled") {
+            return undefined;
+        }
+
+        if (!this.isNewHover(document, position)) {
+            if (this.ready) {
+                const contentMarkdown = new vscode.MarkdownString(`$(sparkle) Copilot\n\n${this.content}\n\n_${localize("copilot.disclaimer", "AI-generated content may be incorrect.")}_`, true);
```

Comment:
* Why does `AI-generated content may be incorrect.` need to be shown? Should that already be obvious to our target audience? Can that warning go in the GitHub Copilot docs or something?
* If we still want to show it, can it only be shown once? i.e. a persistent state could be set to stop showing it.
* If we want to show it more than once, can we show it only once per session?

Reply:
This text is included as part of our Responsible AI compliance for this feature. The rules aren't entirely clear so we're erring on the side of caution by always showing the warning. While it may be acceptable to show it less frequently, given that we've already submitted our Responsible AI documentation for review it would introduce additional delays in shipping this feature.

This is roughly the same UX that we had when this feature first shipped in Visual Studio. Since then, we've been able to relax it to only showing the disclaimer in the link tooltip, but we don't have the ability to control the link tooltip in VS Code.

Reply:
Ok, FYI, I don't see the warning text currently shown for other GitHub copilot features I've tried.

Reply:
@benmcmorran You can add it to the hover link via adding ` "text"` after the command link: 
![image](https://github.com/user-attachments/assets/b98179d5-7259-488f-bee8-eb92aa27c35d)

e.g.
```TypeScript
`](command:C_Cpp.ShowCopilotHover "${localize("copilot.disclaimer", "AI-generated content may be incorrect.")}")`
```

Reply:
Thanks, I didn't realize we had that level of control!

Reply:
I've modified the link to show the tooltip text as Sean showed above and currently removed the warning in the final output. This now matches the current VS implementation.